bazel/_gevent_test_main.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2021 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import grpc
+import unittest
+import sys
+import os
+import pkgutil
+
+import zope
+from zope.event import subscribers
+
+class SingleLoader(object):
+    def __init__(self, pattern):
+        self._pattern = pattern
+        self.suite = unittest.TestSuite()
+        self._loader = unittest.TestLoader()
+        tests = []
+        for importer, module_name, is_package in pkgutil.walk_packages([os.path.dirname(os.path.relpath(__file__))]):
+            if self._pattern in module_name:
+                module = importer.find_module(module_name).load_module(module_name)
+                tests.append(self._loader.loadTestsFromModule(module))
```

Comment:
optional: Maybe we can use `TestLoader.discover(pattern=f"{module_name}.py")`?

Reply:
Very nice. I didn't know about this method.

Reply:
Actually, on closer inspection, this doesn't seem to work. I've put it back to the method that I was using before.